components/kubeflow/pytorch-launcher/sample.py

```diff
@@ -0,0 +1,125 @@
+    master = {
+        "replicas": 1,
+        "restartPolicy": "OnFailure",
+        "template": {
+            "metadata": {
+                "annotations": {
+                    # See https://github.com/kubeflow/website/issues/2011
+                    "sidecar.istio.io/inject": "false"
+                }
+            },
+            "spec": {
+                "containers": [
+                    {
+                    # To override default command
+                    # "command": [
+                    #   "python",
+                    #   "/opt/mnist/src/mnist.py"
+                    # ],
+                    "args": [
+                        "--backend", "gloo",
+                    ],
+                    # Or, create your own image from 
+                    # https://github.com/kubeflow/pytorch-operator/tree/master/examples/mnist
+                    "image": "gcr.io/kubeflow-examples/pytorch-dist-mnist:v20180702-a57993c",
+                    "name": "pytorch",
+                    "resources": {
+                        "requests": {
+                            "memory": "4Gi",
+                            "cpu": "2000m",
+                        },
+                        "limits": {
+                            "memory": "4Gi",
+                            "cpu": "2000m",
+                        },
+                    },
+                    # If using GPUs
+                    # "resources": {
+                    #   "limits": {
+                    #     "nvidia.com/gpu": 1,
+                    #   }
+                    # }
+                    }
+                ],
+                # If imagePullSecrets required
+                # "imagePullSecrets": [
+                #     {"name": "image-pull-secret"},
+                # ],
+            },
+        },
+    }
+
+    # You could have a worker spec that is different from master
+    # (maybe master needs a PVC for saving results)
+    # This does not handle replicas==0.  Would need if statement at kfp
+    # runtime to handle that (doing 'if worker_replicas > 0' here does
+    # not work)
+    worker = copy.deepcopy(master)
+    worker['replicas'] = worker_replicas
+
+    # Launch and monitor the job with the launcher
+    pytorchjob_launcher_op(
+        # Note: name needs to be a unique pytorchjob name in the namespace.
+        # Using RUN_ID_PLACEHOLDER is one way of getting something unique.
+        name=f"name-{kfp.dsl.RUN_ID_PLACEHOLDER}",
+        namespace=namespace,
+        master_spec=master,
+        # pass worker_spec as a string because the JSON serializer will convert
+        # the placeholder for worker_replicas (which it sees as a string) into
+        # a quoted variable (eg a string) instead of an unquoted variable
+        # (number).  If worker_replicas is quoted in the spec, it will break in
+        # k8s.  See https://github.com/kubeflow/pipelines/issues/4776
+        worker_spec=str(worker),
```

Comment:
I thought `str()` wraps string values in single quotes, which is not a valid json. Won't you have issues later if you ever deserialize this to an object?
But see above comments for moving the operations into a component.